File: backend/app/homeowner/schemas/product.py
"""Recommended-product schema — mirrors frontend/lib/homeowner/types/product.ts."""

from app.shared.schemas.user import CamelModel


class RecommendedProductOut(CamelModel):
    id: str
    brand_id: str
    brand_name: str
    name: str
    category: str
    price: int | None = None
    image_url: str | None = None
    match_score: int
    match_reason: str

File: frontend/lib/homeowner/types/product.ts
/**
 * Product-recommendation domain type. Backed by (future)
 * project_product_matches — brandId + productId are the join keys the
 * brand-facing analytics dashboard will aggregate impressions / clicks
 * against.
 */

export type ProductCategorySlug =
  | "wall-ovens"
  | "cooktops"
  | "range-hoods"
  | "fans-and-blowers"
  | "fridges"
  | "microwaves"
  | "dishwashers";

export interface RecommendedProduct {
  id: string;
  brandId: string;
  brandName: string;
  name: string;
  category: ProductCategorySlug;
  price: number | null;
  /** Optional first-party image; when null, the UI renders a category icon. */
  imageUrl: string | null;
  /** 0–100 compatibility from the (future) product-matching engine. */
  matchScore: number;
  matchReason: string;
}
